experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch

```diff
@@ -1,21 +1,51 @@
 diff --git a/esm/interface/getAutocompleteSuggestions.js b/esm/interface/getAutocompleteSuggestions.js
-index ac159d132d095a789b966627d315f4238c79ab60..a3be2d27c9d5c7d6fd3745093301a07d28e79a8d 100644
+index ac159d132d095a789b966627d315f4238c79ab60..a5c69725101bdf5919e27cd55c42e231813b0543 100644
 --- a/esm/interface/getAutocompleteSuggestions.js
 +++ b/esm/interface/getAutocompleteSuggestions.js
-@@ -234,7 +234,12 @@ export function getAutocompleteSuggestions(schema, queryText, cursor, contextTok
+@@ -63,6 +63,9 @@ const hasTypeSystemDefinitions = (sdl) => {
+     }
+     return hasTypeSystemDef;
+ };
++const primitives = new Set(['Int', 'Float', 'String', 'Boolean'])
++const propertyDirectives = new Set(['fold', 'recurse', 'optional', 'transform'])
++const edgeDirectives = new Set(['filter', 'tag', 'output', 'transform'])
+ export function getAutocompleteSuggestions(schema, queryText, cursor, contextToken, fragmentDefs, options) {
+     var _a;
+     const opts = Object.assign(Object.assign({}, options), { schema });
+@@ -234,7 +237,19 @@ export function getAutocompleteSuggestions(schema, queryText, cursor, contextTok
          return getSuggestionsForVariableDefinition(token, schema, kind);
      }
      if (kind === RuleKinds.DIRECTIVE) {
 -        return getSuggestionsForDirective(token, state, schema, kind);
 +        // We should autocomplete fields on the line after we autocomplete a directive
-+        if (state.needsAdvance) { // active after '@' or many other kinds of punctuation, so perfect for when we need to autocomplete directives
-+            return getSuggestionsForDirective(token, state, schema, kind);
++        if (state.needsAdvance || state.name === null) { // active after '@' or many other kinds of punctuation, so perfect for when we need to autocomplete directives.
++                                                         // `state` is `null` when the user autocompletes after typing just '@' then opening autocomplete menu
++            // further restrict the directives we suggest based on the type of the field we are on
++            // x.label does not include the '@'
++            if (primitives.has(typeInfo.fieldDef.type.name)) {
```

Comment:
What happens here if the field type is something like `[String!]` or `[[Int]]!`?

Reply:
Amazing point, thank you! Then it will be an object of

```json
{
    "ofType": { // <- the list
        "name": "String" // <- the actual type
    }
}
```

So the solution is